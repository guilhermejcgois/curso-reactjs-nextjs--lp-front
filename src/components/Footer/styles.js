import style, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = style.p`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
