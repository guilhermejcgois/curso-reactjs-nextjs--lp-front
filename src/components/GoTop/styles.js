import styled from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => `
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-contener: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
