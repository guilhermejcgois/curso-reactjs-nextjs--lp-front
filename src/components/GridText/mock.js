export default {
  background: false,
  title: 'My grid',
  description:
    'Etiam a sapien vitae nisi accumsan aliquet. Quisque pellentesque facilisis lacus, eu iaculis ligula cursus eget.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sit amet dui a nisl molestie mollis. Quisque aliquet, purus at commodo tincidunt, sem orci molestie libero, pretium laoreet sapien sem vel dui. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Morbi id rutrum mauris. Nullam condimentum egestas urna, ac auctor turpis faucibus nec. Donec sit amet laoreet libero. Vivamus imperdiet nibh ut mollis placerat. Etiam egestas tempor scelerisque. Phasellus nunc neque, cursus eu orci vel, dictum molestie est.',
    },
  ],
};
