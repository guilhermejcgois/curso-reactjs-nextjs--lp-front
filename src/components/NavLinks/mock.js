export default [
  {
    children: 'link 1',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'link 2',
    link: '#target2',
    newTab: false,
  },
  {
    children: 'link 3',
    link: '#target3',
    newTab: false,
  },
  {
    children: 'link 4',
    link: '#target4',
    newTab: false,
  },
  {
    children: 'link 5',
    link: '#target5',
    newTab: false,
  },
  {
    children: 'link 6',
    link: '#target6',
    newTab: false,
  },
  {
    children: 'link 7',
    link: '#target7',
    newTab: false,
  },
  {
    children: 'link 8',
    link: '#target8',
    newTab: false,
  },
  {
    children: 'link 9',
    link: '#target9',
    newTab: false,
  },
  {
    children: 'link 10',
    link: '#target10',
    newTab: false,
  },
];
