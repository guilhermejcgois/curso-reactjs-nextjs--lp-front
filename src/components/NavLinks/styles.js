import style, { css } from 'styled-components';

export const Container = style.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row-wrap;

    @media ${theme.media.lteMedium} {
      flex-flow: column no-wrap;
      align-content: center;
    }
  `}
`;
