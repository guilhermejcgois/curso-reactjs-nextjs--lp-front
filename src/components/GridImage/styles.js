import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
