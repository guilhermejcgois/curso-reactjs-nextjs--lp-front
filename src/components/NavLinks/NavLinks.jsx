import P from 'prop-types';
import React from 'react';
import * as Styled from './styles';
import { MenuLink } from '../MenuLink';

export const NavLinks = ({ links = [] }) => (
  <Styled.Container aria-label="Main menu">
    {links.map((link) => (
      <MenuLink key={link.link} {...link} />
    ))}
  </Styled.Container>
);

NavLinks.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
};
