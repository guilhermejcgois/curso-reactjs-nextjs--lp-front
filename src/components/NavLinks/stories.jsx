import React from 'react';
import { NavLinks } from './NavLinks';
import links from './mock';

export default {
  title: 'NavLinks',
  component: NavLinks,
  args: {
    links,
  },
  argTypes: {
    links: { type: '' },
  },
};

export const Template = (args) => (
  <div style={{ display: 'flex' }}>
    <NavLinks {...args} />
  </div>
);
