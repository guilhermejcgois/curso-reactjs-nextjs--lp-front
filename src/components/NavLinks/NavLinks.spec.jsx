import React from 'react';
import { screen } from '@testing-library/react';

import { renderTheme } from '../../styles/render-theme';
import { NavLinks } from './NavLinks';
import mock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/)).toHaveLength(0);
  });

  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getByText(/link 10/).parentElement).toHaveStyleRule('flex-flow', 'column no-wrap', {
      media: theme.media.lteMedium,
    });
  });

  it('should match snapshots', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toMatchSnapshot();
  });
});
