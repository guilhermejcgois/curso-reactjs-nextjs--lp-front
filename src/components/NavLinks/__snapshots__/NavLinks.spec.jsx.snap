// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshots 1`] = `
Array [
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target1"
    role="link"
    target="_self"
  >
    link 1
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target2"
    role="link"
    target="_self"
  >
    link 2
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target3"
    role="link"
    target="_self"
  >
    link 3
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target4"
    role="link"
    target="_self"
  >
    link 4
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target5"
    role="link"
    target="_self"
  >
    link 5
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target6"
    role="link"
    target="_self"
  >
    link 6
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target7"
    role="link"
    target="_self"
  >
    link 7
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target8"
    role="link"
    target="_self"
  >
    link 8
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target9"
    role="link"
    target="_self"
  >
    link 9
  </a>,
  .c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {

}

<a
    class="c0"
    href="#target10"
    role="link"
    target="_self"
  >
    link 10
  </a>,
]
`;
